src/apps/js_generic/js_generic.cpp

```diff
@@ -93,34 +88,58 @@ namespace ccfapp
       if (!out_buf)
       {
         js::js_dump_error(ctx);
-        return nullptr;
-      }
-      {
-        const std::lock_guard<std::mutex> lock(bytecode_cache_mutex);
-        bytecode_cache.insert({module_name_kv,
-                               std::make_shared<std::vector<uint8_t>>(
-                                 out_buf, out_buf + out_buf_len)});
+        throw std::runtime_error(fmt::format(
+          "failed to serialize bytecode for module '{}'", module_name));
       }
+      modules_bytecode->put(
+        module_name_kv, std::vector<uint8_t>(out_buf, out_buf + out_buf_len));
       js_free(ctx, out_buf);
     }
     else
     {
       LOG_TRACE_FMT("Loading module from cache '{}'", module_name_kv);
 
-      auto& bytecode = bytecode_cache[module_name_kv];
+      auto bytecode = modules_bytecode->get(module_name_kv);
       module_val = JS_ReadObject(
         ctx, bytecode->data(), bytecode->size(), JS_READ_OBJ_BYTECODE);
       if (JS_IsException(module_val))
       {
         js::js_dump_error(ctx);
-        return nullptr;
+        throw std::runtime_error(fmt::format(
+          "failed to deserialize bytecode for module '{}'", module_name));
       }
       if (JS_ResolveModule(ctx, module_val) < 0)
       {
+        js::js_dump_error(ctx);
         throw std::runtime_error(fmt::format(
-          "Failed to resolve module dependencies for {}", module_name_kv));
+          "failed to resolve dependencies for module '{}'", module_name));
```

Comment:
We capitalise log lines overwhelmingly, I don't think we have a clear policy for exceptions, but I think we ought to try and be consistent at least.